### ISSUE TITLE
 Update azure_* crates to 0.20

### ISSUE THREAD

**Issue by arpad-m** [neondatabase/neon#7545]
PR #7539 has done the 0.19 update but for the 0.20 update we are blocked by https://github.com/Azure/azure-sdk-for-rust/issues/1652 , because 0.20 is pulling in async-io which we don't want in our dependency graph.

**Comment by skyzh** [comment]
might be related: https://github.com/apache/opendal, not sure if they have s3 time travel support but would be good for unifying everything without worrying about vendor-dependent things.

**Comment by arpad-m** [comment]
@skyzh that's interesting. could you file a separate issue about evaluating adoption of opendal? It's orthogonal to this issue.

**Comment by arpad-m** [comment]
There is a 0.21 version now, and we still can't update, because of https://github.com/Azure/azure-sdk-for-rust/issues/1652 .
